src/Surfnet/Stepup/Identity/Value/DocumentNumber.php

```diff
@@ -21,47 +21,54 @@
 use JsonSerializable;
 use Surfnet\Stepup\Exception\InvalidArgumentException;
 
+use function preg_match;
+use function strval;
+
 final class DocumentNumber implements JsonSerializable
 {
     /**
-     * @var string
+     * @var string|null
      */
     private $documentNumber;
 
     /**
      * @return self
      */
-    public static function unknown()
+    public static function unknown(): self
     {
-        return new self('—');
+        return new self(null);
     }
 
     /**
-     * @param string $documentNumber
+     * @param string|null $documentNumber
      */
-    public function __construct($documentNumber)
+    public function __construct(?string $documentNumber)
     {
-        if (!is_string($documentNumber) || empty($documentNumber)) {
+        if ($documentNumber === null) {
+            // Created using the static ::unknown method
```

Comment:
Some work remains to be done here, also see my previous comment about a forgotten Identity.

Reply:
This is a defense in depth measure. Since we do not know what the format is of the document numbers – these are whatever the issuer decided to put on the document – retroactively setting a limit in the middleware in this way can cause issues when there are existing document numbers in the event stream that do not validate anymore. Limiting what is allowed in is fine, but this should be done on entry. We could do this in the middleware, but as a defense in depth it is best placed in the RA when the form post is received, not in the middleware.

SInce we do generate the activation codes, we know exactly what they look like so we could validate them in the middleware without additional risk.

Reply:
I did that, but Peter didn't want it there :-/